packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -336,6 +339,15 @@ export class RunningSummarizer implements IDisposable {
                     timePending: Date.now() - this.heuristics.lastAttempted.summaryTime,
                 });
             });
+        // back-compat 0.34 noSetPendingAckTimerTimeoutCallback
+        if (summaryCollection.setPendingAckTimerTimeoutCallback !== undefined) {
+            summaryCollection.setPendingAckTimerTimeoutCallback(maxAckWaitTime, () => {
+                if (this.pendingAckTimer.hasTimer) {
+                    this.pendingAckTimer.clear();
+                    this.trySummarize("summaryAckMiss");
+                }
+            });
+        }
```

Comment:
Can use coalescing if you want:
```suggestion
        summaryCollection.setPendingAckTimerTimeoutCallback?.(maxAckWaitTime, () => {
            if (this.pendingAckTimer.hasTimer) {
                this.pendingAckTimer.clear();
                this.trySummarize("summaryAckMiss");
            }
        });
```